doc/hardware/peripherals/sensor/index.rst

```diff
@@ -1,231 +1,108 @@
-example, threshold values for interrupts.  These values are configured via
-attributes.  The example in the following section showcases a sensor with an
-interrupt line that is triggered when the temperature crosses a threshold.
-The threshold is configured at runtime using an attribute.
-
-Triggers
-********
-
-:dfn:`Triggers` in Zephyr refer to the interrupt lines of the sensor chips.
-Many sensor chips support one or more triggers. Some examples of triggers
-include: new data is ready for reading, a channel value has crossed a
-threshold, or the device has sensed motion.
-
-To configure a trigger, an application needs to supply a
-:c:struct:`sensor_trigger` and a handler function.  The structure contains the
-trigger type and the channel on which the trigger must be configured.
-
-Because most sensors are connected via SPI or I2C buses, it is not possible
-to communicate with them from the interrupt execution context.  The
-execution of the trigger handler is deferred to a thread, so that data
-fetching operations are possible.  A driver can spawn its own thread to fetch
-data, thus ensuring minimum latency. Alternatively, multiple sensor drivers
-can share a system-wide thread. The shared thread approach increases the
-latency of handling interrupts but uses less memory. You can configure which
-approach to follow for each driver. Most drivers can entirely disable
-triggers resulting in a smaller footprint.
-
-The following example contains a trigger fired whenever temperature crosses
-the 26 degree Celsius threshold. It also samples the temperature every
-second. A real application would ideally disable periodic sampling in the
-interest of saving power. Since the application has direct access to the
-kernel config symbols, no trigger is registered when triggering was disabled
-by the driver's configuration.
-
-.. literalinclude:: ../../../../samples/sensor/mcp9808/src/main.c
-   :language: c
-   :lines: 12-
-   :linenos:
-
-.. _sensor_api_reference:
+   Getting samples from sensors using the sensor API today can be done in one of
+   two ways. A stable and long lived API :ref:`sensor-fetch-and-get` or a newer
+   but rapidly stabilizing API :ref:`sensor-read-and-decode`. It's expected that
+   in the near future that :ref:`sensor-fetch-and-get` will be deprecated in
+   favor of :ref:`sensor-read-and-decode`. Triggers are handled entirely
+   differently for :ref:`sensor-fetch-and-get` or :ref:`sensor-read-and-decode`
+   and the differences are noted in each of those sections.
+
+.. toctree::
+   :maxdepth: 1
+
+   attributes.rst
+   channels.rst
+   triggers.rst
+   power_management.rst
+   device_tree.rst
+   fetch_and_get.rst
+   read_and_decode.rst
+
+
+.. _sensor-implementing:
+
+Implementing Sensor Drivers
+***************************
+
+Implementing the driver side of the Sensor API requires an understanding first of
+how the sensor APIs are used by reading through :ref:`sensor-using` so do start
+there!
```

Comment:
```suggestion
.. note:: 
   It is recommended to first get a good understanding of
   :ref:`how to use the Sensor API <sensor-using>` before looking into implementing a sensor driver.
```

Reply:
I didn't use this exact wording, but did re-word this and made it a .. note: